lib/rubocop/cop/style/hash_syntax.rb

```diff
@@ -96,7 +96,21 @@ def word_symbol_pair?(pair)
           return false unless key.sym_type?
 
           sym_name = key.loc.expression.source
-          sym_name !~ /\A:["']|=\z/
+          return false if ends_with_equals? sym_name
+          valid_19_syntax_symbol? sym_name
+        end
+
+        def valid_19_syntax_symbol?(sym_name)
```

Comment:
Maybe a good approach would be to try to parse `"{ #{sym_name}: :foo }"` – if it parses then `sym_name` is valid and if it doesn’t then it’s not?